clientv3/client.go

```diff
@@ -306,14 +306,18 @@ func (c *Client) dial(endpoint string, dopts ...grpc.DialOption) (*grpc.ClientCo
 			defer cancel()
 			ctx = cctx
 		}
-		if err := c.getToken(ctx); err != nil {
-			if err == ctx.Err() && ctx.Err() != c.ctx.Err() {
-				err = grpc.ErrClientConnTimeout
+
+		err := c.getToken(ctx)
```

Comment:
Could auth return a bogus token from server side instead of expecting the client to know if auth is enabled or not? When auth is turned on the server will reject it and the client will renegotiate. That'll avoid having this extra error handling path.